### PR TITLE
feat: QRCode reader for deep links - WPB-14547

### PR DESCRIPTION
**Pull request by KaterinaWire**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-14547" title="WPB-14547" target="_blank">TaskWPB-14547</a>  [iOS] Add QRCode reader for deep links to the Developer tools
  </td></table>
  <br />
 

### Issue

This was implemented during the platform day.
QR code scanner for deep links in developer tools.

### Testing

Open Developer Tools -> Deep links -> Scan the QR code

---

### Checklist

- [ ] Title contains a reference JIRA issue number like `[WPB-XXX]`.
- [x] Description is filled and free of optional paragraphs.
- [ ] Adds/updates automated tests.

---

### UI accessibility checklist

_If your PR includes UI changes, please utilize this checklist:_
- [ ] Make sure you use the API for UI elements that support large fonts.
- [ ] All colors are taken from WireDesign.ColorTheme or constructed using WireDesign.BaseColorPalette.
- [ ] New UI elements have Accessibility strings for VoiceOver.

### CODE REVIEW

**Review by jullianm — APPROVED** [review]

File: wire-ios/Wire-iOS/Sources/Developer/DeveloperTools/DeepLinks/QRCodeScannerViewController.swift [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+            failed()
+            return
+        }
+
+        previewLayer = AVCaptureVideoPreviewLayer(session: captureSession)
+        previewLayer.frame = view.layer.bounds
+        previewLayer.videoGravity = .resizeAspectFill
+        view.layer.addSublayer(previewLayer)
+
+        captureSession.startRunning()
```
Comment (netbe) [inline wire-ios/Wire-iOS/Sources/Developer/DeveloperTools/DeepLinks/QRCodeScannerViewController.swift]: -[AVCaptureSession startRunning] should be called from background thread. Calling it on the main thread can lead to UI unresponsiveness

**Review by netbe — COMMENTED** [review]
I found there's one runtime warning to fix, and @KaterinaWire could you add a ticket to it please?

**Comment by KaterinaWire** [comment]
> I found there's one runtime warning to fix, and @KaterinaWire could you add a ticket to it please?

@netbe Created

File: wire-ios/Wire-iOS/Sources/Developer/DeveloperTools/DeepLinks/QRCodeScannerViewController.swift (hunk 1 above)
Comment (KaterinaWire) [reply wire-ios/Wire-iOS/Sources/Developer/DeveloperTools/DeepLinks/QRCodeScannerViewController.swift]: Added:
```
DispatchQueue.global(qos: .userInitiated).async {
            self.captureSession.startRunning()
        }
```

**Review by netbe — APPROVED** [review]

**Pull request merged**